### ISSUE TITLE
regression in fragment function

### ISSUE THREAD

**Issue by bartex80** [secdev/scapy#415]
The fragment function is broken since v2.3.3.
Test case:
```python
p = fragment(Ether() / IP() / UDP())[0]
assert p[IP].proto == 17
assert Ether(str(p))[IP].proto == 17  # bug!
```

**Comment by p-l-** [comment]
Thanks for reporting this, and thank you very much for providing the simple test case!
I have added your test case to the regression tests so hopefully we won't break that again.

Can you check and tell us if your (probably more complicated) real cases work with the patch from #416? Thanks!

**Comment by bartex80** [comment]
I can confirm that the patch fixes real cases too.
Thanks

**Comment by p-l-** [comment]
Fixed by #416.
